frontend/src/components/datasets/ShareGroupDatasetModal.tsx

```diff
@@ -0,0 +1,117 @@
+import React, {useEffect, useState} from "react";
+
+import {useDispatch, useSelector} from "react-redux";
+import { Autocomplete, Button, Container, Dialog, DialogActions, DialogContent, DialogTitle, Divider, FormControl, InputLabel, MenuItem, Select, TextField, Typography } from "@mui/material";
+import {fetchGroups} from "../../actions/group";
+import {RootState} from "../../types/data";
+import {setDatasetGroupRole} from "../../actions/dataset";
+import {useParams} from "react-router-dom";
+
+type ShareGroupDatasetModalProps = {
+    open: boolean,
+    handleClose: any,
+    datasetName: string
+}
+
+export default function ShareGroupDatasetModal(props: ShareGroupDatasetModalProps) {
+    const { open, handleClose, datasetName } = props;
+	const {datasetId} = useParams<{ datasetId?: string }>();
+    const [role, setRole] = useState("viewer");
+	const [group, setGroup] = useState("");
+	const dispatch = useDispatch();
+	const listGroups = () => dispatch(fetchGroups(0, 21));
+	const groups = useSelector((state: RootState) => state.group.groups);
+	const setGroupRole = (datasetId: string , groupId: string, role: string) => dispatch(setDatasetGroupRole(datasetId, groupId, role));
+
+
+	// component did mount
+	useEffect(() => {
+		listGroups();
+	}, []);
+
+    const onShare = () => {
+    	console.log(group, datasetId,role);
+    	setGroupRole(datasetId, group, role);
+        handleClose();
+    }
+
+
+
+	const options = Array();
+    groups.map((group) => {
+    	let group_option = {value:group.id, label:group.name}
+		options.push(group_option);
+    });
+	console.log('group optioins are', options);
```

Comment:
I've reviewed this PR and tested it, looks good to me! Two minor comments:

1. Can we remove these console log statements?
2. We need to enforce that only the dataset "owner" can share roles

I'll add the remaining changes to this branch directly so we can  merge both features together. Thanks!

Reply:
We might need to think about whether an editor should be able to share a dataset. But that can be a separate issue.